### ISSUE TITLE
[TimezonePicker] 'inputProps' does not override search behavior.

### ISSUE THREAD

**Issue by arsenush** [palantir/blueprint#4353]
#### Steps to reproduce
```javascript
        <TimezonePicker
          ...
          inputProps={{
            value: this.searchQuery,
            onChange: this.handlerWhichShouldBeCalledButItIsnt
          }}
        >
```
#### Actual behavior

Callback for search input change is not called.

#### Expected behavior

According to docs callback for search input change should override Select's behavior and should be called.

<img width="1232" alt="Screenshot 2020-10-01 at 00 13 34" src="https://user-images.githubusercontent.com/25638740/94741020-2e25f680-037c-11eb-8eca-8652119ccb96.png">

#### Possible solution

The `onQueryChange` prop which you pass to `Select` in `timezonePicker.tsx` is just:
```javascript
private handleQueryChange = (query: string) => this.setState({ query });
```
No mentions about `inputProps.onChange`.

And as I understand in `Select` `onQueryChange` has some priority over `inputProps.onChange`.
Thus my callback which I pass to `TimezonePicker` is not called.

But it's just an assumption.

Please let me know if I do something wrong...

**Comment by adidahiya** [comment]
Yeah, it looks like this API is unsupported despite what the docs say there. I would welcome a PR to add this feature.

Can you share a bit more about your use case and why you're trying to use a custom input change callback?

**Comment by arsenush** [comment]
I am showing a custom right icon in the input when the search query is present (and hide it when it's empty).
